src/log.rs

```diff
@@ -125,18 +125,49 @@ unsafe extern "C" fn log_callback(
                 )
             });
             let gobject_type_value = gobject_type_value.as_deref();
-            let gstobject_name = gobject_with_ty.and_then(|(obj, ty)| unsafe {
+            let gstobject = gobject_with_ty.and_then(|(obj, ty)| unsafe {
                 // SAFETY: g_type_is_a is provided valid types.
                 if bool::from_glib(g_type_is_a(ty, gst_object_get_type())) {
                     let gstobject = obj as *mut gstreamer::ffi::GstObject;
-                    // SAFETY: GstObject type has been verified above, `name` can be null and we
-                    // check for it. It "should" be valid null-terminated string if not null,
-                    // however.
-                    Some(CStr::from_ptr((*gstobject).name.as_ref()?).to_string_lossy())
+
+                    Some(gstobject)
                 } else {
                     None
                 }
             });
+
+            let gstobject_name = gstobject.as_ref().and_then(|gstobject| unsafe {
+                // SAFETY: GstObject type has been verified above, `name` can be null and we
+                // check for it. It "should" be valid null-terminated string if not null,
+                // however.
+                Some(CStr::from_ptr((*(*gstobject)).name.as_ref()?).to_string_lossy())
+            });
+
+            let user_span = gstobject.as_ref().and_then(|gstobject| unsafe {
+                let data_key = CString::new("tracing-gstreamer:span").unwrap();
```

Comment:
This allocates a string on the heap every time we're in this area. Since the string is statically known, I’d say just write `b"tracing-gstreamer:span\0"`. Should be much faster if nothing else.